src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/IntermediaryMinecraftProvider.java

```diff
@@ -58,9 +59,37 @@ public MergedImpl(Project project, MergedMinecraftProvider minecraftProvider) {
 
 		@Override
 		public List<RemappedJars> getRemappedJars() {
-			return List.of(
-				new RemappedJars(minecraftProvider.getMergedJar(), getMergedJar(), MappingsNamespace.OFFICIAL)
-			);
+			return minecraftProvider.canMergeJars()
+				? List.of(
+					new RemappedJars(minecraftProvider.getMergedJar(), getMergedJar(), MappingsNamespace.OFFICIAL)
+				) : List.of(
+					new RemappedJars(minecraftProvider.getClientJar(), getClientJar(), MappingsNamespace.CLIENT_OFFICIAL),
+					new RemappedJars(minecraftProvider.getServerJar(), getServerJar(), MappingsNamespace.SERVER_OFFICIAL)
+				);
+		}
+
+		@Override
+		public List<MinecraftJar> provide(ProvideContext context) throws Exception {
+			List<MinecraftJar> jars = super.provide(context);
+
+			if (minecraftProvider.canMergeJars()) {
+				return jars;
+			}
```

Comment:
Again, this feels wrong as you are in IntermediaryMinecraftProvider.Merged but getting two none merged jars. Using two of the eixsting SingleJar's should achieve the intended result?

Reply:
I'm not sure there is really another way here. The obfuscated jars cannot be merged so you need to do the jar merging *after* the intermediary mapped jars have been provided. Even if I added new mapped minecraft provider impls for this, I'd still need to do the merging here.